pkgs/development/libraries/qt-5/modules/qtbase.nix

```
@@ -161,6 +171,11 @@ stdenv.mkDerivation (finalAttrs: {
     export MAKEFLAGS+=" -j$NIX_BUILD_CORES"
 
     ./bin/syncqt.pl -version $version
+  '' + lib.optionalString (stdenv.buildPlatform != stdenv.hostPlatform) ''
+    # QT's configure script will refuse to use pkg-config unless these two environment variables are set
+    export PKG_CONFIG_SYSROOT_DIR=/
+    export PKG_CONFIG_LIBDIR=${lib.getLib pkg-config}/lib
+    echo "QMAKE_PKG_CONFIG=''$''${CROSS_COMPILE}pkg-config" >> mkspecs/devices/${qtPlatformCross stdenv.hostPlatform}/qmake.conf
```

Built the change. This gives me a broken result in the final qmake.conf:

```
QMAKE_PKG_CONFIG=1{CROSS_COMPILE}pkg-config
```

because this is quoted with `"` and [the shell expands the `$$` to its pid](https://www.gnu.org/software/bash/manual/html_node/Special-Parameters.html#index-_0024) during the echoing.

```suggestion
    echo 'QMAKE_PKG_CONFIG=''$''${CROSS_COMPILE}pkg-config' >> mkspecs/devices/${qtPlatformCross stdenv.hostPlatform}/qmake.conf
```

This gives me the correct result:

```
QMAKE_PKG_CONFIG=$${CROSS_COMPILE}pkg-config
```

---

I hate bash.